Makefile

```diff
@@ -20,6 +20,8 @@ wasm:
 	wasm-pack build --release --features=wasm
 	cp webhid.js pkg/
 	jq '.files += ["webhid.js"]' pkg/package.json > tmp.json && mv tmp.json pkg/package.json
+	$(eval npm_version=$(shell cat NPM_VERSION))
```

Comment:
why `eval`? can't you just do `npm_version=$(cat NPM_VERSION)`?

what happens if NPM_VERSION contains a newline (code editors often insert a newline at the end)? if it fails in that case it would be good to fix that

Reply:
That does not work as intended, `npm_version` is not set when I assign like that. This was the only solution I could find to make it work. I think it was because of the environment of the makefile, that is different from a shell.
I checked out [this link](https://superuser.com/questions/1752412/variables-not-being-set-in-makefile) to come up with a solution.

Reply:
For new line, I can add one more command to strip the newline character in the end.

Reply:
if you put `npm_version = $(shell cat NPM_VERSION)` at the top of the Makefile, `eval` is not needed. this is probably better as `eval` is always a bit sketchy :smile:

Reply:
about newline: it seems to work as is - maybe `jq` already strips it

Reply:
Okay, I do it like that!